.github/workflows/syntax-checks.yml

```diff
@@ -0,0 +1,72 @@
+name: Syntax Checks
+
+on: [push]
+
+jobs:
+  build-linux:
+    name: CI py${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.5', '3.6', '3.7', '3.8', '3.9']
+      max-parallel: 5
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup Miniconda
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        activate-environment:
+            anaconda-test-env-py-${{ matrix.python-version }}
+        auto-update-conda: true
+        channel-priority: true
+        channels: defaults,conda-forge
+        # environment-file: environment.yml
+        mamba-version: "*"
+        python-version: ${{ matrix.python-version }}
+        use-only-tar-bz2: true  # required for caching
+    - shell: bash -l {0}
+      run: |
+        conda info
+        conda list
+        conda config --show-sources
+        conda config --show
+        conda env list
+        printenv | sort
+
+    # $CONDA is an environment variable pointing to the
+    # root of the miniconda directory
+
+    - name: Install dependencies
+      run: |
+        $CONDA/bin/conda env update --file environment.yml --name $ENV_NAME
+        # $CONDA/bin/conda install pytest --name $ENV_NAME
+        $CONDA/bin/conda list -r --name $ENV_NAME
+      env:
+        PY_VER: ${{ matrix.python-version }}
+        ENV_NAME: anaconda-test-env-py-${{ matrix.python-version }}
+
+    - name: Run tests
+      run: |
+        source $CONDA/bin/activate $ENV_NAME
+        # pytest -vvv ./utils/test_suite.py
+        python ./utils/test_suite.py
+      env:
+        PY_VER: ${{ matrix.python-version }}
+        ENV_NAME: anaconda-test-env-py-${{ matrix.python-version }}
+
+    - name: Build the documentation
+      run: |
+        source $CONDA/bin/activate $ENV_NAME
+        make makebuilddir
+        pushd build/manual
+        # CI needs latexmk to execute next line
```

Comment:
I found on WSL2 with Ubunutu 20.04.1 that I needed these packages to build the PDFs
```
python3-lxml
python-is-python3
python3-pip
xsltproc
texlive-latex-extra
latexmk
xindy
```
then used
```
$ pip3 install --user sphinx
```
And added `~/.local/bin` to `PATH`

Reply:
I'd like to try and keep the setup to using pip rather than the ubuntu-specific packages.  This would make transition to other build OS more consistent.

Reply:
But, some things, such as `texlive-latex-extra` and `latexmk` are unavoidable.  What is `xindy`?

Reply:
I think one of the style files used by sphinx required it.

Reply:
Let's keep that in mind when/if we change the CI instructions to use the OS-provided toolset and not *conda*.  Since the docs build in full with no errors now using the conda toolset, it's not necessary to use the OS-provided python toolset.

Reply:
Do we need to build the PDF documentation during these syntax checks?  We're not publishing from this workflow.  Can we leave out this step until we decide to publish the docs from this workflow (related to #663)?